pkg/bpf/perf.go

```diff
@@ -88,7 +88,7 @@ int perf_event_read(void *_state, void *buf, void *_msg, void *_sample, void *_l
 	struct read_state *state = _state;
 	struct event_sample *e = state->begin;
 
-	if (state->begin == state->end)
+	if (state->begin >= state->end)
```

Comment:
This  is the fix, rest is only stuff for test.

Reply:
If we wrap around, wouldn't `state->begin + e->header.size` be bigger but not `state->begin` itself?

Reply:
@nebril How reproducible is this, do you have some more debugging data?

Reply:
I understand now. I mistook for `state->end` pointing to the end of the ring buffer but that's not the case at all.